src/calculateAverage.js

```diff
@@ -37,7 +37,28 @@ function applyCalculateAverage() {
    * @param {Object} options - optional
    */
   [].__proto__.calculateAverage = function(options) {
-    // write code here
+    if (options === undefined) {
+      return this.reduce((sum, item) => item + sum, 0) / this.length;
+    }
+
+    if (options === null) {
+      return undefined;
+    }
+
+    if (options.propertyName) {
+      const value = options.propertyName;
+      const listOfOptions = this.filter(item => item[value]
+        && !isNaN(item[value]));
+
+      return listOfOptions.length === 0
+        ? undefined
+        : listOfOptions.reduce((sum, item) => sum + item[value], 0)
```

Comment:
will be great to reuse it, but not neccessary

Reply:
Can you explain this more detail, please?

Reply:
I talk about some function `getAverage(arr) => average`, which you can reuse in 2 -3 places, but it is not neccessary